onnxruntime/core/external_ops/pyop.cc

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifdef _DEBUG
+#undef _DEBUG
+#include <Python.h>
+#define _DEBUG
+#else
+#include <Python.h>
+#endif
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+#include "numpy/arrayobject.h"
+#include <functional>
+#include <iostream>
+#include <sstream>
+#include <numeric>
+#include <vector>
+#include <memory>
+#include <mutex>
+#include <functional>
+#include <unordered_map>
+
+using namespace std;
+namespace onnxruntime {
+
+#ifdef _WIN32
+#define PYOP_EXPORT extern "C" __declspec(dllexport)
+#else
+#define PYOP_EXPORT extern "C"
+#endif
+
+struct Finalizer
+{
+    ~Finalizer() {
+        Py_Finalize();
+    }
+};
+
+class Scope
+{
+public:
+    Scope(const vector<PyObject*>& objs = {}): objs_(objs) {
+        mtx_.lock();
+    }
+    ~Scope() {
+        for (auto obj: objs_) {
+            Py_XDECREF(obj);
+        }
+        mtx_.unlock();
+    }
+    void Add(PyObject* obj) {
+        objs_.push_back(obj);
+    }
+private:
+    static std::mutex mtx_;
+    vector<PyObject*> objs_;
+};
+
+std::mutex Scope::mtx_;
+
+PYOP_EXPORT bool Initialize() 
+{
+    Scope scope;
+    Py_Initialize();
+    if (_import_array() < 0) {
+        return false;
+    }
+    auto path_list = PySys_GetObject("path");// do not release it
+    if (nullptr == path_list || !PyList_Check(path_list) ||
+        PyList_Append(path_list, PyUnicode_FromString(".")) != 0) {
+        return false; 
+    }
+    static Finalizer finalizer;
```

Comment:
why static?

Reply:
Because we want to unload the Python C API library on process exit for only once.